codice/ddf
# DDF-2344 Pull out functionality from CatalogFrameworkImpl to logical subcomponents.

#### What does this PR do?
This is the first phase of a change to `CatalogFrameworkImpl` which turns it into a facade over several logical delegate classes. These delegates silo logically like functionality in order to facilitate further refactoring, cleanup, and future maintenance.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@rzwiefel @kcwire @ryeats @tbatie 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@shaundmorris
@stustison

#### How should this be tested?
As this breaks apart one of the core components of DDF, it will require not only a full build with itests, but an extensive run through manual use of the system to ensure that no functionality has been broken.

#### Any background context you want to provide?
The current version of `CatalogFrameworkImpl` in master is 3397 lines long and contains much duplicative, redundant code. It is difficult to maintain and rather fragile. By breaking it up, we can isolate logical silos of work that are easier to reason about, test, and maintain. They will also be more easily refactored to remove duplication and inefficient code.

#### What are the relevant tickets?
DDF-2344

#### Screenshots (if appropriate)
N/A

#### Checklist:
